### ISSUE TITLE
Merging generate-package-note variant 

### ISSUE THREAD

**Issue by ismailhkose** [systemd/package-notes#13]
Hello,

**TLDR;**
I have an implementation of [generate-package-note](https://github.com/microsoft/CBL-Mariner/blob/dev/SPECS/mariner-rpm-macros/generate-package-note.py) tool, and we have been already using it in the internal projects for a while. I want to merge our tool here to have single version  of the tool in open source community. I hope it could be okay to contribute here and combine our version with yours. I wonder if you could accept my changes in your repo, or what changes you could accept here.

**Background story**
We were discussing different format at the beginning to embed meta data into ELF files.  @bluca, who was part of the discussion [here](https://github.com/systemd/systemd/issues/18433), were helping us to communicate with  your community while defining the format. He advised us to follow the JSON format you have agreed. Therefore we could spread this format across the open source community easier.

The internal version of the script, which is now publicly available at [CBL-Mariner](https://github.com/microsoft/CBL-Mariner) github repository, already follows the JSON format discussed [here](https://github.com/systemd/systemd/issues/18433), and [here in the documentation](https://systemd.io/COREDUMP_PACKAGE_METADATA/). It has some additional fields for our needs.

It generates same output file that was described in [this discussion:](https://github.com/systemd/systemd/issues/18433#issuecomment-794279985).

There are design differences between scripts, and it includes additional meta data, which is optionally passed to the tool at command line. Also, It supports generating two types of outputs. 

1 - Linker script and optionally generated header file. Mariner project now uses this method.

2 - C code that keeps .note.package section in an array and mapped to .note.package note section. This method will be depricated soon, as soon as the dependent patches applied to our binutils version. Older version of binutils have Linker script related bug, and it can't get note section information thru linker script. I'm trying to find the relevant patches to backport.

Before creating a PR against your repo, I wanted to get your opinions about what you think and what could be your preferred merging strategy. The license will be also updated to follow yours.

Please let me know thoughts.

@keszybz, @bluca, @viccie30 , FYI

Thank,
Ismail

**Comment by keszybz** [comment]
Let's close the issue, since there isn't anything to track on this side. A pull request would be welcome.
